api/task_create.go

```diff
@@ -21,89 +21,49 @@ func (h *TaskLifeCycleHandler) CreateTask(w http.ResponseWriter, r *http.Request
 
 	// Decode the task request
 	var req taskRequest
-	requestID := requestIDFromContext(r.Context())
+	ctx := r.Context()
+	requestID := requestIDFromContext(ctx) // TODO: log with request ID
```

Comment:
@findkim the logger should already log with the request ID. The middleware adds it to the context logger which is extracted on line 19. Let me know if this is what you were looking to do?

Reply:
thanks for taking a look and double checking here! Sorry for the vague TODO comment, I was hoping to capture whether the request ID logging is propagated throughout the server-side beyond the handler. Do you know if this is already the case?

Reply:
No worries, I'm going through it now to understand the changes since they'll effect the remainder of the task lifecycle work. They look great so far! I can tell a lot of thought and effort went into it and I'm excited to use these changes in my Create Task CLI :D

I see what you mean, no they aren't propagated down yet, I imagine we could pass the logger via context throughout the server side though, which would be even easier now with your refactor. I recall seeing a comment somewhere in the consul code that initializing logger via context was a TODO for their code base as well.